Sub/Src/ComputerVision/libs/pose_calculation.py

```diff
@@ -0,0 +1,126 @@
+'''
+Copyright 2019, Mohammad Shafi, All rights reserved
+
+Author: Mohammad Shafi <ma.shafi99@gmail.com>
+'''
+import os
+import sys
+import cv2
+import numpy as np
+from MechOS import mechos
+
+PARAM_PATH = os.path.join("..", "Params")
+sys.path.append(PARAM_PATH)
+MECHOS_CONFIG_FILE_PATH = os.path.join(PARAM_PATH, "mechos_network_configs.txt")
+from mechos_network_configs import MechOS_Network_Configs
+
+class distance_calculator():
+    '''
+    This class initializes our required matrices using points from the parameter server,
+    and detection data from yolo. It will then use openCV operations to interpret the data
+    and give us accurate judgements on how far our objects are from the front of the sub.
+    '''
+    def __init__(self, detection, x, y, w, h):
+        '''
+        Initializes my distance calculator
+        Params:
+            detection: Detection object. Includes image with yolo's bouding box drawn around
+            x: x coordinate of the bounding box
+            y: y coordinate of the bounding box
+            w: width of the bounding box
+            h: height of the bounding box
+        Returns:
+            N/A
+        '''
+        self.detection = detection
+        self.x_coordinate = float(x)
+        self.y_coordinate = float(y)
+        self.width = float(w)
+        self.height = float(h)
+        self.camera_matrix = None
+        self.distortion_matrix = None
+        self.two_dim_points = None
+        self.three_dim_points = None
+
+        configs = MechOS_Network_Configs(MECHOS_CONFIG_FILE_PATH)._get_network_parameters()
+
+        self.param_serv = mechos.Parameter_Server_Client(configs["param_ip"], configs["param_port"])
+        self.param_serv.use_parameter_database(configs["param_server_path"])
+
+    def set_matrices(self):
+        '''
+        This function sets our camera, distortion, and dimension points depending on the detection
+        Params:
+            N/A
+        Returns:
+            N/A
+        '''
+
+        self.focal_length_x = float(self.param_serv.get_param("Vision/solvePnP/camera_matrix/focal_length_x"))
+        self.focal_length_y = float(self.param_serv.get_param("Vision/solvePnP/camera_matrix/focal_length_y"))
+        self.optical_center_x = float(self.param_serv.get_param("Vision/solvePnP/camera_matrix/optical_center_x"))
+        self.optical_center_y = float(self.param_serv.get_param("Vision/solvePnP/camera_matrix/optical_center_y"))
```

Comment:
I assume the parameter server works fast, but if its slow at all then doing all this loading for every detection for every frame might bog things down a bit.